doonut-app-external-api/src/main/kotlin/com/doonutmate/oauth/configuration/AuthorizationArgumentResolver.kt

```diff
@@ -0,0 +1,44 @@
+package com.doonutmate.oauth.configuration
+
+import com.doonutmate.oauth.JwtTokenProvider
+import com.doonutmate.oauth.exception.BaseException
+import com.doonutmate.oauth.exception.BaseExceptionCode
+import jakarta.servlet.http.HttpServletRequest
+import org.springframework.core.MethodParameter
+import org.springframework.stereotype.Component
+import org.springframework.web.bind.support.WebDataBinderFactory
+import org.springframework.web.context.request.NativeWebRequest
+import org.springframework.web.method.support.HandlerMethodArgumentResolver
+import org.springframework.web.method.support.ModelAndViewContainer
+
+@Component
+class AuthorizationArgumentResolver(
+    private val jwtTokenProvider: JwtTokenProvider,
+) : HandlerMethodArgumentResolver {
+    companion object {
+        const val BEARER_PREFIX_LEN = 7
+    }
+
+    override fun supportsParameter(parameter: MethodParameter): Boolean {
+        return parameter.getParameterAnnotation(Authorization::class.java) != null
+    }
```

Comment:
[MethodParameter.hasParameterAnnotation()](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/core/MethodParameter.html#hasParameterAnnotation(java.lang.Class)) 메서드를 사용하면 코드가 더 깔끔해질 것 같아요.  

```suggestion
    override fun supportsParameter(parameter: MethodParameter): Boolean {
        return parameter.hasParameterAnnotation(Authorization::class.java)
    }
```